pkg/apis/core/validation/validation.go

```diff
@@ -1669,6 +1681,42 @@ func ValidationOptionsForPersistentVolume(pv, oldPv *core.PersistentVolume) Pers
 		// If the old object allowed "ReadWriteOncePod", continue to allow it in the new object
 		opts.AllowReadWriteOncePod = true
 	}
+	// if its CSI spec and secretName was DNS1035Label format for oldPV
```

Comment:
These blocks are hard to review and vulnerable to copy/paste errors, and is doing unnecessary evaluation if one of the earlier blocks already set opts.AllowDNS1123SubDomainSecrets to `true`. Can this be structured something like this instead:

```go
if oldCSI := oldPv.Spec.CSI; oldCSI != nil {
  opts.AllowDNS1123SubDomainSecrets = 
    secretRefRequiresSubdomainSecretName(oldCSI.ControllerExpandSecretRef) ||
    secretRefRequiresSubdomainSecretName(oldCSI.ControllerPublishSecretRef) ||
    secretRefRequiresSubdomainSecretName(oldCSI.NodeStageSecretRef) ||
    secretRefRequiresSubdomainSecretName(oldCSI.NodePublishSecretRef)    
  }
}
...

func secretRefRequiresSubdomainSecretName(secretRef *core.SecretReference) bool {
  // ref and name were specified and name didn't fit within label validation
  return secretRef != nil && len(secretRef.Name) > 0 && len(validation.IsDNS1123Label(secretRef.Name)) > 0
}
```

Reply:
agreed.  rearranged to follow above format :+1: